### PR TITLE
✨ Introduce kubestellar-list-syncing-objects

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR adds a demo/utility command that lists or watches the objects of a given kind in the mailbox workspaces.

Example usage:
```
kubestellar-list-mid-dynamic --api-group apps --api-kind ReplicaSet
```

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
/cc @waltforme 
/cc @clubanderson 
@yana1205

**Comment by clubanderson** [comment]
I like the idea, absolutely hate the name.  Can we make something that is super simple here?  simply "kubestellar-monitor-resources" would be great.

**Comment by MikeSpreitzer** [comment]
I do not love the name either.
I put "mid" in the name suggesting "middle" because it is specifically looking in a middle type of place, _between_ the WECs and the WDSes. The objects of interest also appear in the WECs, and in the WDSes in the 1:1 case. So I thought it important that the name be specific to the fact that it is looking in the middle.

This command is ambidextrous: it can produce a one-shot listing, and it can do on-going monitoring. I expect that for your new quickstarts the one-shot listing will be the more interesting case. There is some precedent for taking the "on-going" part as a modulation of "list": `kubectl get` can get a list of objects, and takes an optional command line flag that turns it into an on-going watch.

File: cmd/kubestellar-list-mid-dynamic/main.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+	if len(kind) == 0 {
+		logger.Error(nil, "The --kind must not be the empty string")
+	}
+	if len(gvr.Resource) == 0 {
+		gvr.Resource = strings.ToLower(kind) + "s"
+	}
+
+	parentClientConfig, err := parentClientOpts.ToRESTConfig()
+	if err != nil {
+		logger.Error(err, "failed to make ESPW client config")
```
Comment (waltforme) [inline cmd/kubestellar-list-mid-dynamic/main.go:94]: Maybe change from 'ESPW' to 'parent' here as well?

File: cmd/kubestellar-list-mid-dynamic/main.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	parentInformerFactory := kcpinformers.NewSharedScopedInformerFactory(parentClient, 0, "")
+	mbPreInformer := parentInformerFactory.Tenancy().V1alpha1().Workspaces()
+
+	informer := mailboxwatch.NewSharedInformer[k8sdynamic.NamespaceableResourceInterface, *unstructured.UnstructuredList](ctx, listGVK, mbPreInformer, dynamicClusterResource, exampleObj, 0, upstreamcache.Indexers{})
+
+	informer.AddEventHandler(upstreamcache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj any) { log(logger, "add", obj) },
+		UpdateFunc: func(oldObj, newObj any) { log(logger, "update", newObj) },
+		DeleteFunc: func(obj any) { log(logger, "delete", obj) },
+	})
```
Comment (waltforme) [inline cmd/kubestellar-list-mid-dynamic/main.go:132]: It is fun and creative to me that, logic other than enqueuing are written here in the ResourceEventHandlerFuncs. I like it 👍

**Comment by MikeSpreitzer** [comment]
The force-push to 1004e8dab fixes import ordering and the editing oversight called out in review above.

**Comment by clubanderson** [comment]
Let's talk this through, while this command works, it will be lost in obscurity if it is not renamed.

**Comment by MikeSpreitzer** [comment]
I am open to renaming. I mentioned some of my concerns above. The "dynamic" part is descriptive for Kubernetes experts but is perhaps not something that needs to be in the name. Following are some suggestions.

- `kubestellar-list-middle-objects`
- `kubestellar-list-objects-in-middle`
- `kubestellar-list-full-objects`

**Comment by clubanderson** [comment]
how about 'kubestellar-list-objects-pending-sync'

File: cmd/kubestellar-list-mid-dynamic/main.go (hunk 1 above)
Comment (MikeSpreitzer) [reply cmd/kubestellar-list-mid-dynamic/main.go]: Thanks, fixed.

**Comment by MikeSpreitzer** [comment]
Remember that an object subject to downsync has state flowing in two directions: the desired state flows from core to WEC, and the reported state flows from WEC to core.

A mailbox space holds both objects subject to downsync and objects subject to upsync.

These objects appear there regardless of whether any propagation remains to be done, so "pending" is not quite on-point.

**Comment by MikeSpreitzer** [comment]
An object in a mailbox space may already have been synced.  Yes, qualifying with "pending sync" and "to sync" have the same issue: they imply that the list will only have objects that have not yet been synced.

**Comment by MikeSpreitzer** [comment]
How about even simpler, `kubestellar-list-syncing-object` ?

**Comment by clubanderson** [comment]
sold

**Comment by MikeSpreitzer** [comment]
The force-push to 25d8d279c is a rebase plus the agreed renaming plus adding documentation.

**Pull request merged**